src/NuGet.Core/NuGet.Protocol/Resources/ReadmeUriTemplateResource.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+#if NETCOREAPP
+using System;
+#endif
+
+namespace NuGet.Protocol
+{
+    /// <summary>
+    /// A resource that provides the URI for downloading a README file based on a template.
+    /// </summary>
+    internal class ReadmeUriTemplateResource : INuGetResource
+    {
+        private readonly string _uriTemplate;
+
+        public ReadmeUriTemplateResource(string uriTemplate)
+        {
+            _uriTemplate = uriTemplate;
+        }
+
+        /// <summary>
+        /// Get  the URL for downloading the readme file.
+        /// </summary>
+        /// <param name="id">The package id</param>
+        /// <param name="version">The package version</param>
+        /// <returns>URL to download README, built using the URI template.</returns>
+        public string GetReadmeUrl(string id, NuGetVersion version)
+        {
+            if (_uriTemplate == null)
+            {
+                return string.Empty;
+            }
+
+            var uriString = _uriTemplate
+#if NETCOREAPP
+               .Replace("{lower_id}", id.ToLowerInvariant(), StringComparison.OrdinalIgnoreCase)
+               .Replace("{lower_version}", version.ToNormalizedString().ToLowerInvariant(), StringComparison.OrdinalIgnoreCase);
+#else
+               .Replace("{lower_id}", id.ToLowerInvariant())
+               .Replace("{lower_version}", version.ToNormalizedString().ToLowerInvariant());
```

Comment:
It would be good to make the tokens constants so we don't duplicate them and in case of future changes we make sure they're changed in only one place.